From ec69a1ac63bb78e3d196153c12724fb8cbcb92d1 Mon Sep 17 00:00:00 2001
From: Raviteja Narayanam <raviteja.narayanam@xilinx.com>
Date: Mon, 9 Nov 2020 15:45:10 +0530
Subject: [PATCH 1754/1852] regulator: da9121: Modify the variant code as per
 hardware

commit 0775d1d0aab5964c017e60a83a3496b7ddb7675a from
https://github.com/Xilinx/linux-xlnx.git

The ICs provided to xilinx have the following variant codes,
DA9130 - 0X00
DA9131 - 0X01
Updated the same in the driver.

Signed-off-by: Raviteja Narayanam <raviteja.narayanam@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/regulator/da9121-regulator.c | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/regulator/da9121-regulator.c b/drivers/regulator/da9121-regulator.c
index 7757d2ba4d29..2d8ac636a50e 100644
--- a/drivers/regulator/da9121-regulator.c
+++ b/drivers/regulator/da9121-regulator.c
@@ -40,8 +40,8 @@
 /* Device ID list */
 #define DA9121_DEVICE_ID	0x05
 #define DA9121_VARIANT_MRC_BASE	0x2
-#define DA9121_VARIANT_VRC	0x1
-#define DA9220_VARIANT_VRC	0x0
+#define DA9130_VARIANT_VRC	0X0
+#define DA9131_VARIANT_VRC	0X1
 #define DA9122_VARIANT_VRC	0x2
 #define DA9217_VARIANT_VRC	0x7
 
@@ -1201,15 +1201,11 @@ static int da9121_get_device_type(struct i2c_client *i2c, struct da9121 *chip)
 	variant_vrc = variant_id & DA9121_MASK_OTP_VARIANT_ID_VRC;
 
 	switch (variant_vrc) {
-	case DA9121_VARIANT_VRC:
+	case DA9130_VARIANT_VRC:
 		type = "DA9121/DA9130";
 		device_config_match = ( chip->variant_id == DA9121_TYPE_DA9121_DA9130);
 		break;
-	case DA9220_VARIANT_VRC:
-		type = "DA9220/DA9132";
-		device_config_match = ( chip->variant_id == DA9121_TYPE_DA9220_DA9132);
-		break;
-	case DA9122_VARIANT_VRC:
+	case DA9131_VARIANT_VRC:
 		type = "DA9122/DA9131";
 		device_config_match = ( chip->variant_id == DA9121_TYPE_DA9122_DA9131);
 		break;
-- 
2.31.1

